### PR TITLE
Add functionality for faculty to view and filter applications

### PR DESCRIPTION
**Pull request by hpitawela**
This PR includes functionality implementation, rspec and BDD for Application filtering by faculty. Faculty can filter applications by research interests, CGPA, GRE-Verbal, GRE-Quantitative, undergraduate university and PhD/MS degree objective.
This PR contain commits from @hpitawela and @Avanti0532 .

## Issue Related
#4 #6

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Front end change

## Code coverage
95.14 %

## Screenshot (if any change in front end)
Screen Shot 2019-03-23 at 4 47 21 PM

### CODE REVIEW

File: app/assets/stylesheets/sInterestedSchools.css [hunk 1]
```diff
@@ -30,7 +30,7 @@ div#main {
 }
 .trash{
     position: absolute;
-    right: 410px;
+    /*right: 410px;*/
```
Comment (Avanti0532) [inline app/assets/stylesheets/sInterestedSchools.css]: Please remove this if not required.

File: features/ApplicationFilterByFaculty.feature [hunk 2]
```diff
@@ -0,0 +1,123 @@
+Feature: Faculty can look at all applications applied to his university
```
Comment (Avanti0532) [inline features/ApplicationFilterByFaculty.feature]: I would suggest you to add following scenarios:
1.  Faculty can filter applications by GPA range
2.  Faculty can filter applications by GRE quant and verbal score

File: app/assets/javascripts/yeti/research_interests.js [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+    });
+
+    $('#doneModal').click(function (e) {
+        e.stopImmediatePropagation();
+        var array_interests_ids = [];
+        var array_interests_labels = [];
+        var no_checked = $("#researchInterestsModal input:checkbox:checked").length;
+        console.log(no_checked);
+        if (no_checked == 0){
+            console.log('In IF');
```
Comment (linhpha) [inline app/assets/javascripts/yeti/research_interests.js]: Please remove logging

File: app/assets/stylesheets/sInterestedSchools.css (hunk 1 above)
Comment (hpitawela) [reply app/assets/stylesheets/sInterestedSchools.css]: Removed.

File: features/ApplicationFilterByFaculty.feature (hunk 2 above)
Comment (hpitawela) [reply features/ApplicationFilterByFaculty.feature]: Added cucumber testcases.

File: app/assets/javascripts/yeti/research_interests.js (hunk 3 above)
Comment (hpitawela) [reply app/assets/javascripts/yeti/research_interests.js]: Removed.

**Pull request merged**